tests/mock_network/churn.rs

```diff
@@ -419,6 +410,70 @@ impl RelocationMapBuilder {
     }
 }
 
+// When do we send messages.
+#[derive(PartialEq, Eq)]
+enum MessageSchedule {
+    AfterChurn,
+    DuringChurn,
+}
+
+fn progress_and_verify<R: Rng>(
+    rng: &mut R,
+    network: &Network,
+    nodes: &mut [TestNode],
+    message_schedule: MessageSchedule,
+    added_indices: BTreeSet<usize>,
+    dropped_names: BTreeSet<XorName>,
+) {
+    if !dropped_names.is_empty() {
+        warn!("Dropping {:?}", dropped_names);
+    }
+
+    let (expectations, relocation_map) = match message_schedule {
+        MessageSchedule::AfterChurn => {
+            poll_and_resend(nodes);
+            let added_names = check_added_indices(nodes, added_indices);
+            log_churn_outcome(&added_names, &dropped_names);
+
+            let expectations = setup_expectations(rng, nodes, network.elder_size());
+            poll_and_resend(nodes);
+
+            (expectations, BTreeMap::default())
+        }
+        MessageSchedule::DuringChurn => {
+            let expectations = setup_expectations(rng, nodes, network.elder_size());
+            let relocation_map = RelocationMapBuilder::new(&nodes);
+
+            poll_and_resend(nodes);
+            let added_names = check_added_indices(nodes, added_indices);
+            log_churn_outcome(&added_names, &dropped_names);
+
+            (expectations, relocation_map.build(&nodes))
+        }
+    };
+
+    expectations.verify(nodes, &relocation_map);
+    verify_invariant_for_all_nodes(network, nodes);
+    shuffle_nodes(rng, nodes);
+}
+
+fn log_churn_outcome(added_names: &BTreeSet<XorName>, dropped_names: &BTreeSet<XorName>) {
+    if !added_names.is_empty() {
+        if !dropped_names.is_empty() {
+            warn!(
+                "Simultaneously added {:?} and dropped {:?}",
```

Comment:
I am not sure this is so helpful, much easier to follow similar sequence in logs.
I find it much easier to look for Dropped line followed by`Added`line, or added and dropped on same line but using same pattern.

so that seem that 

```
fn log_churn_outcome(added_names: &BTreeSet<XorName>, dropped_names: &BTreeSet<XorName>) {
   warn!("Added {:?}, dropped {:?}", added_names, dropped_names);
}
```

Would be simpler and more effective for me?